test/queue-test.js

```diff
@@ -1297,6 +1297,41 @@ describe('Queue', (it) => {
       t.is(err.message, `Job ${job.id} timed out (10 ms)`);
     });
 
+    it('clears a waiting job that is missing data', async (t) => {
+      const queue = t.context.makeQueue();
+
+      const job = await queue.createJob({foo: 'first'}).save();
+
+      // Delete the job's data from redis
+      await new Promise((resolve) =>
+        queue.client.hdel(queue.toKey('jobs'), job.id, resolve)
+      );
```

Comment:
As a general rule, I'd like to handle errors (yes, even in tests - just in case something strange breaks). Probably easiest to just use `callAsync` from `promise-callbacks` (this'll need an additional import clause).
```suggestion
      await callAsync((done) => queue.client.hdel(queue.toKey('jobs'), job.id, done));
```